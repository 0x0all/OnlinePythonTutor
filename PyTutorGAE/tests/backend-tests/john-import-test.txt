import math
import operator
import string
import collections
import re
import json

# these don't seem allowed by bdb in Python 2.6, so punt for now
#import random
#import datetime
#import functools
