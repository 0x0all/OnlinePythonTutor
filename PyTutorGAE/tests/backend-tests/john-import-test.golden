{
  "code": "import math\nimport operator\nimport string\nimport collections\nimport re\nimport json\n\n# these don't seem allowed by bdb in Python 2.6, so punt for now\n#import random\n#import datetime\n#import functools\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "math", 
        "__package__"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "math": [
          "REF", 
          1
        ], 
        "__package__": null
      }, 
      "heap": {
        "1": [
          "module", 
          "<module 'math' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/math.so'>"
        ]
      }, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "math", 
        "__package__", 
        "operator"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "operator": [
          "REF", 
          2
        ], 
        "math": [
          "REF", 
          1
        ], 
        "__package__": null
      }, 
      "heap": {
        "1": [
          "module", 
          "<module 'math' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/math.so'>"
        ], 
        "2": [
          "module", 
          "<module 'operator' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/operator.so'>"
        ]
      }, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "math", 
        "__package__", 
        "operator", 
        "string"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "operator": [
          "REF", 
          2
        ], 
        "string": [
          "REF", 
          3
        ], 
        "math": [
          "REF", 
          1
        ], 
        "__package__": null
      }, 
      "heap": {
        "1": [
          "module", 
          "<module 'math' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/math.so'>"
        ], 
        "2": [
          "module", 
          "<module 'operator' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/operator.so'>"
        ], 
        "3": [
          "module", 
          "<module 'string' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/string.pyc'>"
        ]
      }, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "math", 
        "__package__", 
        "operator", 
        "string", 
        "collections"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "operator": [
          "REF", 
          2
        ], 
        "collections": [
          "REF", 
          4
        ], 
        "string": [
          "REF", 
          3
        ], 
        "math": [
          "REF", 
          1
        ], 
        "__package__": null
      }, 
      "heap": {
        "1": [
          "module", 
          "<module 'math' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/math.so'>"
        ], 
        "2": [
          "module", 
          "<module 'operator' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/operator.so'>"
        ], 
        "3": [
          "module", 
          "<module 'string' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/string.pyc'>"
        ], 
        "4": [
          "module", 
          "<module 'collections' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/collections.pyc'>"
        ]
      }, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "math", 
        "__package__", 
        "operator", 
        "string", 
        "collections", 
        "re"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "string": [
          "REF", 
          3
        ], 
        "__package__": null, 
        "re": [
          "REF", 
          5
        ], 
        "collections": [
          "REF", 
          4
        ], 
        "operator": [
          "REF", 
          2
        ], 
        "math": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "module", 
          "<module 'math' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/math.so'>"
        ], 
        "2": [
          "module", 
          "<module 'operator' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/operator.so'>"
        ], 
        "3": [
          "module", 
          "<module 'string' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/string.pyc'>"
        ], 
        "4": [
          "module", 
          "<module 'collections' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/collections.pyc'>"
        ], 
        "5": [
          "module", 
          "<module 're' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/re.pyc'>"
        ]
      }, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "math", 
        "__package__", 
        "operator", 
        "string", 
        "collections", 
        "re", 
        "json"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "string": [
          "REF", 
          3
        ], 
        "__package__": null, 
        "re": [
          "REF", 
          5
        ], 
        "json": [
          "REF", 
          6
        ], 
        "collections": [
          "REF", 
          4
        ], 
        "operator": [
          "REF", 
          2
        ], 
        "math": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "module", 
          "<module 'math' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/math.so'>"
        ], 
        "2": [
          "module", 
          "<module 'operator' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/lib-dynload/operator.so'>"
        ], 
        "3": [
          "module", 
          "<module 'string' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/string.pyc'>"
        ], 
        "4": [
          "module", 
          "<module 'collections' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/collections.pyc'>"
        ], 
        "5": [
          "module", 
          "<module 're' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/re.pyc'>"
        ], 
        "6": [
          "module", 
          "<module 'json' from '/Library/Frameworks/Python.framework/Versions/7.0/lib/python2.7/json/__init__.pyc'>"
        ]
      }, 
      "line": 6, 
      "event": "return"
    }
  ]
}
